api/client/src/main/java/org/projectnessie/client/auth/oauth2/JwtToken.java

```diff
@@ -0,0 +1,216 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.client.auth.oauth2;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import javax.annotation.Nullable;
+
+/**
+ * A minimal representation of a JWT token, as defined in <a
+ * href="https://datatracker.ietf.org/doc/html/rfc7519">RFC 7519</a>.
+ *
+ * <p>The current implementation only supports extracting the payload claims. It does not decode nor
+ * validate the header and the signature. It is not particularly efficient, but is suitable for
+ * access by multiple threads.
+ */
+class JwtToken {
+
+  public static final String EXP_CLAIM = "exp";
+  public static final String ISS_CLAIM = "iss";
+  public static final String SUB_CLAIM = "sub";
+  public static final String AUD_CLAIM = "aud";
+  public static final String NBF_CLAIM = "nbf";
+  public static final String IAT_CLAIM = "iat";
+  public static final String JTI_CLAIM = "jti";
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  /**
+   * Parses a JWT token from a string.
+   *
+   * @param token The JWT token to parse.
+   * @return The parsed {@link JwtToken}.
+   * @throws IllegalArgumentException if the token is invalid.
+   */
+  public static JwtToken parse(String token) {
+    if (token != null) {
+      @SuppressWarnings("StringSplitter")
+      String[] parts = token.split("\\.");
+      if (parts.length == 3 && parts[1].length() > 0) {
+        try {
+          JsonNode payload = OBJECT_MAPPER.readTree(Base64.getUrlDecoder().decode(parts[1]));
+          return new JwtToken(payload);
+        } catch (IOException ignored) {
+          // fall-through
+        }
+      }
+    }
+    throw new IllegalArgumentException("Invalid JWT token: " + token);
+  }
+
+  private final JsonNode payload;
+  private final ConcurrentMap<String, Object> claims;
+
+  private JwtToken(JsonNode payload) {
+    this.payload = payload;
+    claims = new ConcurrentHashMap<>();
+  }
+
+  /** Returns the raw payload of the JWT token. */
+  public JsonNode getPayload() {
+    return payload;
+  }
+
+  /**
+   * The "iss" (issuer) claim identifies the principal that issued the JWT. The processing of this
+   * claim is generally application specific. The "iss" value is a case-sensitive string containing
+   * a StringOrURI value. Use of this claim is OPTIONAL.
+   */
+  @Nullable
+  public String getIssuer() {
+    return (String)
+        claims.computeIfAbsent(
+            ISS_CLAIM,
+            k ->
+                payload.has(ISS_CLAIM) && payload.get(ISS_CLAIM).isTextual()
```

Comment:
Would something like this make it overall simpler?

```java

  private <T> T payloadFieldValue(String field, Predicate<JsonNode> tester, Function<JsonNode, T> mapper) {
    JsonNode value = payload.get(field);
    if (value == null || !tester.test(value)) {
      return null;
    }
    return mapper.apply(value);
  }
```

and for this `getIssuer`:
```java
          k -> payloadFieldValue(ISS_CLAIM, JsonNode::isTextual, JsonNode::asText));
```